contracts/src/v0.8/automation/dev/v2_3/AutomationRegistryLogicB2_3.sol

```diff
@@ -526,17 +535,22 @@ contract AutomationRegistryLogicB2_3 is AutomationRegistryBase2_3 {
    * @dev this will be deprecated in a future version in favor of getMinBalance
    */
   function getMinBalanceForUpkeep(uint256 id) public view returns (uint96 minBalance) {
-    return getMaxPaymentForGas(_getTriggerType(id), s_upkeep[id].performGas);
+    Upkeep memory upkeep = s_upkeep[id];
+    return getMaxPaymentForGas(_getTriggerType(id), upkeep.performGas, upkeep.billingToken);
   }
 
   /**
    * @notice calculates the maximum payment for a given gas limit
    * @param gasLimit the gas to calculate payment for
    */
-  function getMaxPaymentForGas(Trigger triggerType, uint32 gasLimit) public view returns (uint96 maxPayment) {
```

Comment:
i don't think this may be possible but just want to bring it up: what if a billing token is so small in value that its maxPayment will be larger than uint96.

Reply:
this is on my to-do list... but I think we have to change to uint256 for balances 😅